lib/rspec/support/encoded_string.rb

```diff
@@ -2,7 +2,19 @@ module RSpec
   module Support
     # @private
     class EncodedString
-      MRI_UNICODE_UNKOWN_CHARACTER = "\xEF\xBF\xBD"
+      if String.method_defined?(:encoding)
+        # see https://github.com/ruby/ruby/blob/ca24e581ba/encoding.c#L1191
+        def self.pick_encoding(source_a, source_b)
+          Encoding.compatible?(source_a, source_b) || Encoding.default_external
+        end
+      else
+        def self.pick_encoding(_source_a, _source_b)
+        end
+      end
+
+      # Ruby's default replacement string for is U+FFFD ("\xEF\xBF\xBD") for Unicode encoding forms
+      #   else is '?' ("\x3F")
+      REPLACE = "\x3F"
```

Comment:
Can this just be `REPLACE = "?"`?  I find the `\x` escaped form to be much harder to read (and it winds up requiring a comment to explain it's a question mark).

Also why is this better than the default replacement string of `U+FFFD`?  I don't understand that from your comment.